docs/dqx/docs/reference/quality_rules.mdx

```diff
@@ -2234,6 +2234,119 @@ Using DQX classes:
 
 When using dataset-level checks, the filter condition is applied before aggregation, ensuring that the check operates only on the relevant subset of rows rather than on the aggregated results.
 
+## Converting checks between formats
+
+In DQX, checks can be defined either as Python classes or YAML declarations. When using YAML, the files are first parsed into dictionaries and then transformed into DQX class instances under the hood. Since both formats share the same internal structure, they are fully interchangeable and can be safely converted between one another.
+
+The examples below show how to convert checks between Python class definitions and their equivalent dictionary metadata representations.
+
+### Converting DQX Classes to Metadata
+
+Converting a list of class-based check definitions into their equivalent dictionary representations (metadata format):
+
+<Tabs>
+  <TabItem value="Python" label="Python" default>
+    ```python
+    import yaml
+    import pyspark.sql.functions as F
+    from databricks.labs.dqx.engine import DQEngine
+    from databricks.labs.dqx.rule import DQRowRule, DQDatasetRule
+    from databricks.labs.dqx import check_funcs
+
+    # Define quality checks using DQX classes
+    checks = [
+        DQRowRule(
+            name="a_is_null_or_empty",
+            check_func=check_funcs.is_not_null_and_not_empty,
+            column="a",
+        ),
+        DQRowRule(
+            criticality="warn",
+            check_func=check_funcs.is_not_null_and_is_in_list,
+            column=F.col("c"),
+            check_func_kwargs={"allowed": ["a", F.col("d")]},
+        ),
+        DQDatasetRule(
+            criticality="error",
+            check_func=check_funcs.is_unique,
+            columns=["col1"],
+            check_func_kwargs={"row_filter": "col2 > 0"}
+        ),
+        DQDatasetRule(
+            criticality="error",
+            check_func=check_funcs.is_unique,
+            columns=["col2"],
+            check_func_args=[False, "col2 > 0"]
+        ),
+    ]
+
+    # Convert the checks to dictionary-based metadata format
+    checks_dict: list[dict] = DQEngine.serialize_checks(checks)
+    print(yaml.safe_dump(checks_dict))
+    ```
+  </TabItem>
+</Tabs>
+
+### Converting Metadata to DQX Classes
+
+Converting a list of DQX dictionary representations (metadata format) to DQX classes:
+
+<Tabs>
+  <TabItem value="Python" label="Python" default>
+    ```python
+    import pyspark.sql.functions as F
+    from databricks.labs.dqx.engine import DQEngineCore
+    from databricks.labs.dqx.rule import DQDatasetRule, DQRule
+    from databricks.labs.dqx import check_funcs
+
+    # Convert metadata into DQX class-based check definitions
+    checks_dict = [
```

Comment:
It may be confusing that we showcase dict but talk about yaml before. Should this define the checks using yaml and then convert to dict instead for clarity?

`checks_dict = yaml.safe_load("""....""")`